cypress/integration/my_patches.js

```diff
@@ -1,23 +1,46 @@
 /// <reference types="Cypress" />
 
 const MY_PATCHES_ROUTE = "/my-patches";
-describe("my patches page", function () {
-  beforeEach(() => {
-    cy.server();
+describe("My Patches Page", () => {
+  before(() => {
     cy.login();
   });
 
-  it("Should render with Show Commit Queue box checked when commitQueue not indicated in URL query param", () => {
-    cy.visit(MY_PATCHES_ROUTE);
-    cy.get("[data-cy=commit-queue-checkbox]").should("be.checked");
+  beforeEach(() => {
+    cy.preserveCookies();
+    cy.listenGQL();
   });
 
-  it("Should render with Show Commit Queue box unchecked when commitQueue is false in URL query param", () => {
-    cy.visit(`${MY_PATCHES_ROUTE}?commitQueue=false`);
-    cy.get("[data-cy=commit-queue-checkbox]").should("not.be.checked");
+  it("Initial load should make a userPatches gql query with default query params when none are specified in the URL", () => {
+    cy.visit(MY_PATCHES_ROUTE);
+    cy.waitForGQL("userPatches", {
+      "request.body.variables.$includeCommitQueue": (v) => v == true,
+      "request.body.variables.$limit": (v) => v === 10,
+      "request.body.variables.$page": (v) => v === 0,
+      "request.body.variables.$patchName": (v) => v === "",
+      "request.body.variables.$statuses": (v) =>
+        Array.isArray(v) && v.length === 0,
+    }).then((xhr) => console.log(xhr));
```

Comment:
i don't know if testing the shape of the request so strictly is a good or practical way to test. Testing the end result of the query and how it appears and behaves to the user is the goal of the test imo. we should not need to inspect the request body because testing the usability of the page as a result of it does that job and does it more effectively. 

How Kent C Dodds describes what to test is "Think less about the code you are testing and more about the use cases that code supports."

These tests tests an underlying and private functionality of the feature. Therefore it does not test how the software is _used_.

Reply:
True e2e tests are not possible until the feature is complete. I agree this test should be replaced by a test that asserts the rows rendered in the table after user interaction with the filter inputs.